pcp/PCPrecord_actions.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+# Executed by systemd service 'PCPrecord.service'
+# See: /etc/systemd/system/PCPrecord.service
+################################################################
+
+# GLOBALS ###################
+# Include the PCP Functions file
+source $PWD/pcp_functions.inc
+
+FIFO="/tmp/pcpFIFO"                 # get from cmdline
+sample_rate=5                       # hardcode DEFAULT for now
+pmlogger_running="false"            # Initialize service as OFF
+om_workload_file="/tmp/openmetrics_workload.txt"
+om_workload_file_reset="/tmp/openmetrics_workload_reset.txt"
+
+#############################
+# Functions #################
+
+reset_om_metrics() {
+    echo "Reset OpenMetrics values"
+    cp ${om_workload_file_reset} ${om_workload_file}
+}
+
+error_exit() {
+    if [ "$?" != "0" ]; then
+        systemd-notify --status="ERROR: $1"
+        # Additional error handling logic can be added here
+        rm -f "$FIFO"
+        # Reset openmetric.workload metric values prior to leaving
+        reset_om_metrics
+## if pmlogger_running = True then attempt forcible STOP?
+        exit 1
+    fi
+}
+# END Functions #################
+
+# Main #################
+# Initialize openmetric.workload metric values
+reset_om_metrics
+
+# Verify required files and Packages are available
+#----------------------------------
+test -f "${om_workload_file}"
+error_exit "Initialization: ${om_workload_file} not found!"
+
+# Remove and recreate FIFO on every service 'start'
+rm -f "$FIFO"
+mkfifo "$FIFO"
+error_exit "Initialization: mkfifo $FIFO failed"
+
+## DEBUG - measure processing interval: $postaction-$preaction
+action='NONE'
+interval=0.0
+
+# Infinite Loop  #################
+# Read FIFO and perform requested ACTION (start, stop, ...)
+# Access each word in $action string for parsing 'actions' & 'metric'
+# NOTE: 'Start, Stop, Reset' actions have no metrics
+while : ; do
+    # Required or we get TIMEOUT on 'read action < "$FIFO" '
+    # Signal readiness for next $action. SYNC point w/client Workload
+    # Report timing interval for most recent ACTION
+    systemd-notify --ready --status="READY: last-action - $action = ${interval}ms"
+    # Read the Request/'$action' and then process it
+    read action < "$FIFO"       # Blocks until data is available
+    # Signal busy Processing this $action
+    systemd-notify --status="$action PMLOGGER Request"
+    action_arr=($action)        # Array of 'words' in Request read from FIFO
+## DEBUG - measure processing interval for ACTION: $postaction-$preaction
+    preaction=$(mark_ms)
+    case "${action_arr[0]}" in
+        Start)     # 'Start $archive_dir $test_name $conf_file' 
+            archive_dir="${action_arr[1]}"
```

Comment:
have the case sections call a function when there is numerous lines involved.
Makes it easier to read the case statement.